actionview/lib/action_view/helpers/form_options_helper.rb

```diff
@@ -593,6 +599,25 @@ def time_zone_options_for_select(selected = nil, priority_zones = nil, model = :
         zone_options.safe_concat options_for_select(convert_zones[zones], selected)
       end
 
+      # Returns a string of option tags for the days of the week.
+      #
+      # Options:
+      # * <tt>:index_as_value</tt> - Defaults to false, set to true to use the index of the weekday as the value.
+      # * <tt>:day_format</tt> - The I18n key of the array to use for the weekday options.
+      # Defaults to :day_names, set to :abbr_day_names for abbreviations.
+      #
+      # NOTE: Only the option tags are returned, you have to wrap this call in
+      # a regular HTML select tag.
+      def weekday_options_for_select(selected = nil, index_as_value: false, day_format: :day_names)
+        day_names = I18n.translate("date.#{day_format}")
+
+        if index_as_value
+          day_names = day_names.map.with_index { |day, i| [day, i] }
+        end
```

Comment:
Let's condense this a bit and move it right up to the other `day_names =` assignment.

```suggestion
        day_names = day_names.map.with_index.to_h if index_as_value
```

Reply:
Good call